include/deal.II/fe/fe.h

```diff
@@ -1524,15 +1524,17 @@ class FiniteElement : public Subscriptor, public FiniteElementData<dim>
       ReferenceCell::default_combined_face_orientation()) const;
 
   /**
-   * For lines with non-standard line_orientation in 3d, the dofs on lines
-   * have to be permuted in order to be combined with the correct shape
-   * functions. Given a local dof @p index on a line, return the local index,
-   * if the line has non-standard line_orientation. In 2d and 1d there is no
-   * need for permutation, so the given index is simply returned.
+   * Given a local dof @p index on a line and the orientation @p
+   * combined_orientation of that line, return the local dof which accounts for
+   * @p combined_orientation.
+   *
+   * @note In both 1d and all-quadrilateral meshes in 2d all lines have the
+   * standard orientation.
    */
   unsigned int
-  adjust_line_dof_index_for_line_orientation(const unsigned int index,
-                                             const bool line_orientation) const;
+  adjust_line_dof_index_for_line_orientation(
+    const unsigned int  index,
+    const unsigned char combined_orientation) const;
```

Comment:
Did we ever talk about giving these orientations a named type instead of just using plain `unsigned char`? Just a `typedef` in `namespace types`?

Reply:
In the long run, it might actually be nice to use `std::bitset` for this, but that's a separate question.

Reply:
we should definitely use a distinct integer typedef.

I'd prefer not to use `std::bitset<3>` since `sizeof(std::bitset<3>{}) == 1` whereas `sizeof(unsigned char) == 1`. This size matters in 3D since we allocate an array of `6 * n_cells` of these things for fast face orientation lookup.

Reply:
I think you mean `sizeof(std::bitset<3>) == 8`, do you? I agree, we should not waste space for items that are stored many times in a computation, e.g. for every cell/face/edge/vertex, if we know the range of values is limited, and use as narrow a type as possible. While a single data structure considered in isolation might indeed not appear very problematic, in real computations we use many different data fields in tight sequence, and then every data structure matters because everything hot consumes cache space. It can also be expected to become more pressing in the next 5-10 years. Caches (which consist of SRAM cells) are among those things in a processor (CPU, GPU) that are more difficult to reduce in size than arithmetic/logic units. It has been said for a at least a decade that computations are cheap compared to data access, but the most concern is on the caches in the near-term projections as far as I see.

Reply:
Yes, I didn't meant to waste space. 

A better choice may be [std::byte](https://en.cppreference.com/w/cpp/types/byte), which is explicitly defined as a type on which you can do bit arithmetic, but which is not an integer type. My main concern with using `typedef`s to integer types is always that they convert so easily to everything else, and it's easy to make mistakes or forget places. `std::byte` might solve that problem.

Reply:
Yup, I definitely meant `sizeof(std::bitset<3>) == 8`.

In a lot of places we have loops over the orientation indices, or we use them for array access: that might not work with `std::byte` though it would generally be good to prevent as many conversions as possible.